Makefile

```diff
@@ -185,107 +184,246 @@ test-integration: test-integration-aws test-integration-clickhouse test-integrat
+	$(MAKE) -k stop-integration-kafka \
+    ; rc=$$? \
+	$(MAKE) start-integration-kafka
+	sleep 10 # Many services are very slow... Give them a sec..
 endif
 	${MAYBE_ENVIRONMENT_EXEC} cargo test --no-default-features --features "kafka-integration-tests rdkafka-plain" --lib ::kafka:: -- --nocapture
 ifeq ($(AUTODESPAWN), true)
-	${MAYBE_ENVIRONMENT_EXEC} $(CONTAINER_TOOL)-compose stop
+	$(MAKE) -k stop-integration-kafka
 endif
 
+start-integration-loki:
+	$(CONTAINER_TOOL) network create test-integration-loki
+	$(CONTAINER_TOOL) run -d --network=test-integration-loki -p 3100:3100 -v $(PWD)/tests/data:/etc/loki \
+	 --name loki grafana/loki:master -config.file=/etc/loki/loki-config.yaml
+
+stop-integration-loki:
+	$(CONTAINER_TOOL) rm --force loki 2>/dev/null; true
+	$(CONTAINER_TOOL) network rm test-integration-loki 2>/dev/null; true
+
 test-integration-loki: ## Runs Loki integration tests
 ifeq ($(AUTOSPAWN), true)
-	${MAYBE_ENVIRONMENT_EXEC} $(CONTAINER_TOOL)-compose up -d dependencies-loki
-	sleep 5 # Many services are very lazy... Give them a sec...
+	$(MAKE) -k stop-integration-loki \
+    ; rc=$$? \
+	$(MAKE) start-integration-loki
+	sleep 10 # Many services are very slow... Give them a sec..
 endif
 	${MAYBE_ENVIRONMENT_EXEC} cargo test --no-default-features --features loki-integration-tests --lib ::loki:: -- --nocapture
 ifeq ($(AUTODESPAWN), true)
-	${MAYBE_ENVIRONMENT_EXEC} $(CONTAINER_TOOL)-compose stop
+	$(MAKE) -k stop-integration-loki
 endif
 
+start-integration-pulsar:
+	$(CONTAINER_TOOL) network create test-integration-pulsar
+	$(CONTAINER_TOOL) run -d --network=test-integration-pulsar -p 6650:6650 --name pulsar \
+	 apachepulsar/pulsar bin/pulsar standalone
+
+stop-integration-pulsar:
+	$(CONTAINER_TOOL) rm --force pulsar 2>/dev/null; true
+	$(CONTAINER_TOOL) network rm test-integration-pulsar 2>/dev/null; true
+
 test-integration-pulsar: ## Runs Pulsar integration tests
 ifeq ($(AUTOSPAWN), true)
-	${MAYBE_ENVIRONMENT_EXEC} $(CONTAINER_TOOL)-compose up -d dependencies-pulsar
-	sleep 5 # Many services are very lazy... Give them a sec...
+	$(MAKE) -k stop-integration-pulsar \
+    ; rc=$$? \
+	$(MAKE) start-integration-pulsar
+	sleep 10 # Many services are very slow... Give them a sec..
 endif
 	${MAYBE_ENVIRONMENT_EXEC} cargo test --no-default-features --features pulsar-integration-tests --lib ::pulsar:: -- --nocapture
 ifeq ($(AUTODESPAWN), true)
-	${MAYBE_ENVIRONMENT_EXEC} $(CONTAINER_TOOL)-compose stop
+	$(MAKE) -k stop-integration-pulsar
 endif
 
+start-integration-splunk:
+	$(CONTAINER_TOOL) network create test-integration-splunk
+	$(CONTAINER_TOOL) run -d --network=test-integration-splunk -p 8088:8088 -p 8000:8000 -p 8089:8089 \
+     --name splunk timberio/splunk-hec-test:minus_compose
+
+stop-integration-splunk:
+	$(CONTAINER_TOOL) rm --force splunk 2>/dev/null; true
+	$(CONTAINER_TOOL) network rm test-integration-splunk 2>/dev/null; true
+
 test-integration-splunk: ## Runs Splunk integration tests
```

Comment:
This one fails for me. :( 

```
test sinks::splunk_hec::tests::splunk_encode_event_json ... ok
thread 'sinks::splunk_hec::integration_tests::splunk_insert_source' panicked at 'Timed out while waiting', src/test_util/mod.rs:310:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'sinks::splunk_hec::integration_tests::splunk_custom_fields' panicked at 'Timed out while waiting', src/test_util/mod.rs:310:13
thread 'sinks::splunk_hec::integration_tests::splunk_sourcetype' panicked at 'Timed out while waiting', src/test_util/mod.rs:310:13
thread 'sinks::splunk_hec::integration_tests::splunk_insert_message' panicked at 'Timed out while waiting', src/test_util/mod.rs:310:13
thread 'sinks::splunk_hec::integration_tests::splunk_hostname' panicked at 'Timed out while waiting', src/test_util/mod.rs:310:13
thread 'sinks::splunk_hec::integration_tests::splunk_healthcheck' panicked at 'Timed out while waiting', src/test_util/mod.rs:310:13
test sinks::splunk_hec::integration_tests::splunk_sourcetype ... FAILED
test sinks::splunk_hec::integration_tests::splunk_custom_fields ... FAILED
test sinks::splunk_hec::integration_tests::splunk_insert_message ... FAILED
test sinks::splunk_hec::integration_tests::splunk_insert_source ... FAILED
test sinks::splunk_hec::integration_tests::splunk_hostname ... FAILED
test sinks::splunk_hec::integration_tests::splunk_healthcheck ... FAILED
thread 'sinks::splunk_hec::integration_tests::splunk_insert_index' panicked at 'Timed out while waiting', src/test_util/mod.rs:310:13
thread 'sinks::splunk_hec::integration_tests::splunk_insert_many' panicked at 'Timed out while waiting', src/test_util/mod.rs:310:13
thread 'sinks::splunk_hec::integration_tests::splunk_configure_hostname' panicked at 'Timed out while waiting', src/test_util/mod.rs:310:13
test sinks::splunk_hec::integration_tests::splunk_insert_index ... FAILED
test sinks::splunk_hec::integration_tests::splunk_insert_many ... FAILED
test sinks::splunk_hec::integration_tests::splunk_configure_hostname ... FAILED
```

Reply:
@leebenson Any thoughts?

Reply:
Splunk is probably not booting again. I know our splunk tests are failing in CI from time to time - we probably should create an issue to address this explicitly.

I'll try this as well and see if it breaks on my end.